README.md

```diff
@@ -1,3 +1,97 @@
-# dovetail-cdn-viewer-request
+# Dovetail CDN Viewer Request
 
-Viewer request cloudfront function for the Dovetail CDN
+[CloudFront Function](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/cloudfront-functions.html)
+to verify and normalize all requests to the [Dovetail 3 CDN](https://github.com/PRX/Infrastructure/blob/master/cdn/dovetail3-cdn.yml).
+
+This function is tied to the CloudFront `viewer-request` event, verifies the
+request, redirects old/expired urls, and normalizes the requested paths.
+
+This project follows the [standards for PRX services](https://github.com/PRX/docs.prx.org/blob/master/team/Project-Standards.md#services).
+
+![image](https://user-images.githubusercontent.com/1410587/121265887-cb73eb80-c876-11eb-8cd8-7292da09208c.png)
+
+## Configuration
+
+As this is a CloudFront Function, you must hardcode everything instead of any
+configurations. That said, we use a [Custom::CodeFetcher](https://github.com/PRX/Infrastructure/blob/master/cdn/dovetail3-cdn.yml) and a bit of python in CloudFormation to replace values at deploy time.
+
+- `<EXPIRED_REDIRECT_PREFIX>` - the scheme, domain, and optional path to
+  redirect requests back to if your url `?exp=` is expired.
+
+## Usage
+
+### Inputs
+
+The lambda is invoked with a CloudFront Function [viewer-request event](https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/functions-event-structure.html#functions-event-structure-request)
+_(which is **NOT** the same as an edge function viewer-request event!!!)_ of the
+structure:
+
+```js
+{
+  "request": {
+    "method": "GET",
+    "uri": "/<podcast_id>/<episode_guid>/<arrangement_digest>/filename.mp3",
+    "querystring": {
+      "exp": { "value": "99999999" },
+      "force": { "value": "1" }
+    }
+  }
+}
+```
+
+The `uri` may also optionally include a `<feed_id>` token, to indicate this came
+from a non-default feed (such as "adfree"):
+
+```
+"uri": "/<podcast_id>/<feed_id>/<episode_guid>/<arrangement_digest>/filename.mp3"
+```
+
+The `exp` is a (currently optional) epoch seconds timestamp, when this url will
```

Comment:
So if someone takes an expired URL, and removes `exp` it becomes fresh?

Reply:
At the moment, yes.  As the `exp` isn't really about security, but more just making sure the ads you _do_ get are current campaigns we can get paid for.  (You're not listening to stale campaigns).

But #1 in this repo is to start signing these urls back in Dovetail/DovetailRouter.  Which might play more into adfree audio feeds, rather than the `?exp` param.